front/migrations/20240412_force_use_at_iteration.ts

```diff
@@ -0,0 +1,147 @@
+import { assertNever } from "@dust-tt/types";
+import * as _ from "lodash";
+
+import { AgentConfiguration } from "@app/lib/models";
+import {
+  AgentDustAppRunConfiguration,
+  AgentRetrievalConfiguration,
+  AgentTablesQueryConfiguration,
+} from "@app/lib/models";
+import { AgentGenerationConfiguration } from "@app/lib/models";
+import logger from "@app/logger/logger";
+import { makeScript } from "@app/scripts/helpers";
+
+// Fetch all agents, with all generation configs and all actions.
+// Goal is to backfill forceUseAtIteration for all generations and actions.
+// If there is an action, its forceUseAtIteration will be 0 and the generation's will be 1.
+// If there is no action, the generation's will be 0.
+
+const backfillAgentConfigurations = async (execute: boolean) => {
+  const agents = await AgentConfiguration.findAll({
+    where: {
+      instructions: null,
+    },
+    include: [
+      {
+        model: AgentGenerationConfiguration,
+        as: "generationConfiguration",
+      },
+    ],
+  });
+  const generationConfigByAgentId: Record<
+    number,
+    AgentGenerationConfiguration
+  > = {};
+  for (const agent of agents) {
+    generationConfigByAgentId[agent.id] = agent.generationConfiguration;
+  }
+
+  const retrievalConfigs = await AgentRetrievalConfiguration.findAll();
+  const tablesQueryConfigs = await AgentTablesQueryConfiguration.findAll();
+  const dustAppRunConfigs = await AgentDustAppRunConfiguration.findAll();
+
+  const actionsByAgentId: Record<
+    number,
+    (
+      | AgentRetrievalConfiguration
+      | AgentDustAppRunConfiguration
+      | AgentTablesQueryConfiguration
+    )[]
+  > = _.groupBy(
+    [...retrievalConfigs, ...dustAppRunConfigs, ...tablesQueryConfigs],
+    (action) => action.agentConfigurationId
+  );
+
+  const allAgentIds: number[] = Array.from(
+    new Set<number>([
+      ...Object.keys(actionsByAgentId).map(Number),
+      ...Object.keys(generationConfigByAgentId).map(Number),
+    ])
+  );
+
+  const chunks = _.chunk(allAgentIds, 16);
+
+  for (const c of chunks) {
+    for (const aId of c) {
+      const generation = generationConfigByAgentId[aId];
+      const actions = actionsByAgentId[aId] ?? [];
```

Comment:
Since we want to merge my migration and yours, here we need to add something like: 

```
if (execute) {
    const nbFixedIterations = (generation ? 1 : 0) + actions.length
    await AgentConfiguration.update({ maxToolsUseForRun: 1 }, {
    where: {
      id: aId,
    },
  });
}
````

Reply:
(wrote from Github so definitely not reliable)

Reply:
added